### ISSUE TITLE
After logging into the Xiaomi account, system can't find any device or entity. 登录小米账号后找不到任何实体和设备。

### ISSUE THREAD

**Issue by yjwu-leadstec** [al-one/hass-xiaomi-miot#1907]
### Device model / 设备型号

All devices under Xiaomi account

### Component version / 插件版本

0.7.21

### HA core version / HA版本

2024.10.4

### Integrated mode / 集成方式

Automatic (自动模式)

### The problem / 问题详情

In the past few days, I found that all the Xiaomi devices imported through the plugin are offline, around 99 devices. They were functioning normally before. Today, I tried to troubleshoot and found that after entering the Xiaomi account username and password, it prompted that no devices could be found under this account. I have tried restarting HA and the plugin. Still, I cannot find the devices under the account, and the password has not been changed. Please help. Thanks in advance.

最近这几天发现通过插件导入的小米设备全部都离线了，总数有几十个。之前一直都是正常使用。今天尝试排查，发现填入小米的账号密码后，提示在此账号下找不到任何的设备。重启HA和重启插件都尝试过了。还是不能找到账号下的设备，账号也没有改过密码。求帮助。

### Entity attributes / 实体属性

```yaml
light.lemesh_wy0c09_003d_light
```

### Home Assistant Logs / 系统日志

![image](https://github.com/user-attachments/assets/80ea33a3-2713-49e4-aa23-9f8ae823d41b)

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 632, in async_render
    render_result = _render_with_context(self.template, compiled, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 2729, in _render_with_context
    return template.render(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/jinja2/environment.py", line 1304, in render
    self.environment.handle_exception()
  File "/usr/local/lib/python3.12/site-packages/jinja2/environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 1, in top-level template code
TypeError: '>=' not supported between instances of 'NoneType' and 'int'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 749, in async_render_to_info
    render_info._result = self.async_render(  # noqa: SLF001
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 634, in async_render
    raise TemplateError(err) from err
homeassistant.exceptions.TemplateError: TypeError: '>=' not supported between instances of 'NoneType' and 'int'
2024-10-29 14:59:50.888 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Got xiaomi devices for 150545052 failed: {'code': 0, 'message': 'auth err'}
2024-10-29 14:59:50.889 WARNING (MainThread) [custom_components.xiaomi_miot] None device in xiaomi cloud: 150545052
2024-10-29 14:59:51.017 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Got xiaomi devices for 150545052 failed: {'code': 0, 'message': 'auth err'}
2024-10-29 15:00:00.997 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Got xiaomi devices for 150545052 failed: {'code': 0, 'message': 'auth err'}
2024-10-29 15:00:01.002 WARNING (MainThread) [custom_components.xiaomi_miot] None device in xiaomi cloud: 150545052
2024-10-29 15:00:01.190 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Got xiaomi devices for 150545052 failed: {'code': 0, 'message': 'auth err'}
[home-assistant_2024-10-29T07-43-41.278Z.log](https://github.com/user-attachments/files/17553361/home-assistant_2024-10-29T07-43-41.278Z.log)

**Comment by housihai** [comment]
同样的问题

**Comment by lujiashun1** [comment]
一样的问题

**Comment by housihai** [comment]
回滚到0.7.20就正常了
rollback 0.7.21 to solve it

**Comment by yjwu-leadstec** [comment]
没找到升级前的备份，恢复到前几天的备份，然后整个HA都崩了...

**Comment by MBRjun** [comment]
dup of #1899

**Comment by yjwu-leadstec** [comment]
恢复备份应该要等1小时以上，睡一觉发现HA又能重新使用了。回滚到上一个版本能暂时解决。但我总不能以后都不更新这个插件的吧。还是希望尽快找到问题并修复吧。

**Comment by al-one** [comment]
尝试master分支。
```bash
wget -O - https://get.hacs.vip | DOMAIN=xiaomi_miot ARCHIVE_TAG=master bash -
```

**Comment by al-one** [comment]
Fixed in v0.7.22
